File: solutions/python/word-count/1/word_count.py
import re

def count_words(sentence):
    tokens = re.sub(r"\s|[^a-z0-9']|'\B|\B'", r" ",sentence.lower()).split()
    word_counts = {}
    for word in tokens:
        known_words = word_counts.keys()
        if word not in known_words:
            word_counts[word] = 1
        else:
            word_counts[word] += 1
    return word_counts      

File: solutions/javascript/word-count/1/word-count.js
// word Count 

export const countWords = (phrase) => {
  let counts = {}
  phrase.toLowerCase()
        .split(new RegExp("\\s|[^a-z0-9']|'\\B|\\B'"))
        .forEach(word =>
          {if(word.length>0)
            counts[word] = isNaN(counts[word]) ? +1 : counts[word]+1;});
  return counts;
};
